bchain/coins/eth/ethrpc.go

```diff
@@ -1170,9 +1195,36 @@ func (b *EthereumRPC) EthereumTypeGetBalance(addrDesc bchain.AddressDescriptor)
 
 // EthereumTypeGetNonce returns current balance of an address
 func (b *EthereumRPC) EthereumTypeGetNonce(addrDesc bchain.AddressDescriptor) (uint64, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), b.Timeout)
-	defer cancel()
-	return b.Client.NonceAt(ctx, addrDesc, nil)
+
+	var result string
+	var err error
+
+	if b.alternativeSendTxProvider != nil {
+		result, err = b.alternativeSendTxProvider.callHttpStringResult(
+			b.alternativeSendTxProvider.urls[0],
+			"eth_getTransactionCount",
+			addrDesc,
+			"pending",
+		)
+		if err != nil {
```

Comment:
I would at least log error here. If there is some issue, we will not know at all that it is not using the alternative provider. Also, it would be better to avoid duplicate  calls of `b.callRpcStringResult("eth_getTransactionCount", addrDesc, "pending")`. Maybe the code could be structured that you first try to use the alternative provider (if enabled) and then if there is a `result`, the non alternative way is skipped, otherwise it is used.